### ISSUE TITLE
enable_if typedefs are not idiomatically named

### ISSUE THREAD

**Issue by bkietz** [r-lib/cpp11#66]
Names matching `/is_\w+/` are conventionally reserved for `bool_constant`s, but in cpp11 they connote a typedef of `enable_if`, for example (as.hpp:24):

```c++
template <typename T>
using is_integral = typename std::enable_if<
    std::is_integral<T>::value &&
        !std::is_same<typename std::decay<T>::type, bool>::value &&
        !std::is_same<typename std::decay<T>::type, char>::value,
    typename std::decay<T>::type>::type;
```

The prefix `enable_if_` will make it clearer to newcomers that this is an SFINAE helper.

**Comment by jimhester** [comment]
Sounds like a good suggestion to me.

**Comment by bkietz** [comment]
fixed in #72
